File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.cpp
class Solution {
public:
    static bool cmp(vector<int>&l, vector<int>&r){
      return l[1] > r[1]; 
    }
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
      sort(boxTypes.begin(), boxTypes.end(), cmp);
      int tot = 0;
      int box = 0;
      for(auto boxType:boxTypes){
        if(box == truckSize)
          return tot;
        if(box+boxType[0] <= truckSize){
          tot += boxType[0]*boxType[1];
          box += boxType[0];
        }else{
          tot += (truckSize-box)*boxType[1];
          box += truckSize-box;
        }
      }
      return tot;
    }
};

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key = lambda x: -x[1])
        tot = 0
        box = 0
        for numOfBox, unit in boxTypes:
          print(numOfBox, unit)
          if box == truckSize:
            return tot
          if box+numOfBox <= truckSize:
            tot += unit*numOfBox
            box += numOfBox
          else:
            tot += (truckSize - box)*unit
            box += truckSize - box
        return tot
          
            
          
        